### DISCUSSION TITLE
Ability to get dataset as pandas dataframe

### DISCUSSION THREAD

**Discussion by samuell** [featureform/featureform#445] [Ideas]
Perhaps I have missed some functionality in the training data API, but an idea related to #444 that has came up in our discussions with colleagues at Savantic is it would be awesome if it was possible to query out a set of features for a specific time period and get them as pandas dataframe directly.

The use case again is for data scientists early in their experimental work to be able to pull out a suitable size dataset to have a look at the data and play around with it interactively, plotting it etc.

But also for model training, it seems it is often quite practical to have the data as a dataframe.

One additional benefit, especially if having the feature names as columns, is that it makes it easier to not accidentally mix up different features, which seems quite easy to do right now with the row-oriented API for training data.

**Comment by simba-git** [comment]
We have a pandas() method on sources, we could add one on top of other resources. Let me talk through the team on what would be needed to implement this on local mode.
